### DISCUSSION TITLE
USB and Bluetooth emulation and pass-through

### DISCUSSION THREAD

**Discussion by Torinde** [86Box/86Box#2907] [Ideas]
USB pass-through for host devices to appear inside 86box like VirtualBox. joncampbell123/dosbox-x/issues/3210
Bluetooth pass-through like VMware https://kb.vmware.com/s/article/2005315

Related: #3633

**Comment by parkerlreed** [comment]
This is out of the scope for 86box. These usually require guest side applications for the integration, which there is no planned support for.

**Reply by Torinde** [reply]
Guest-side applications? You mean like VirtualBox guest additions USB host controller driver? Yes, a USB driver will be needed - just like driver for any emulated device. So, depending what device 86box chooses to emulate (e.g. some Intel/VIA/etc. chipset or USB chip or the VirtualBox USB device itself) - the same driver will be used (no driver development needed).

After the guest OS has this USB host controller (with passthrough to host hardware devices), then USB joysticks, cameras, scanners, tablets, storage, audio etc. can be connected. Also emulated versions thereof - joysticks/gamepads, touchscreens (WinXP TabletPC edition), etc.

**Reply by Torinde** [reply]
[Virtualbox 7.2.2](https://www.virtualbox.org/wiki/Changelog-7.2#v2):
> USB Webcam: The virtual USB webcam is now part of the open source base package

**Comment by MrPepka** [comment]
This is a good idea, but as stated above, there are no plans to write the driver you're talking about. But maybe someday

**Comment by derchi83** [comment]
Hi,
I wanted to raise that feature request as well. USB Passthrough for devices that are supported under WIN98 would be very helpfull. On the Client side maybe a driver for a virtual USB PCI device or smth. would be required that is passed through to the Host USB.

**Comment by Fubuki0x10DE** [comment]
Hi,
I think USB device passthrough (such as USB flash disk) is a useful function under Windows 2000/Me. On Windows hosts it can use winusb.sys and libusb.dll to implement USB device passthrough.
